core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/KeyStoreManager.java

```diff
@@ -365,6 +576,43 @@ public String getKeyStorePassword(String keyStoreName) throws Exception {
         }
     }
 
+    /**
+     * Get the private key password for the given key store name.
+     *
+     * @param keyStoreName Key store name.
+     * @return Private key password.
+     * @throws SecurityException If there is an error when getting the private key password.
+     */
+    public String getPrivateKeyPassword(String keyStoreName) throws SecurityException {
```

Comment:
At glance, it's better to not to have such public method. Rather check the places which needed this info and see whether we can utilize some existing in the `KeyStoreManager` to cater that.

Reply:
Addressed with https://github.com/wso2/carbon-kernel/pull/4099/commits/f89bbb95d862ca9f626a21592685bbd3e230eb11